test/cpp/end2end/end2end_test.cc

```diff
@@ -588,6 +588,54 @@ TEST_F(End2endTest, RpcMaxMessageSize) {
   EXPECT_FALSE(s.IsOk());
 }
 
+TEST_F(End2endTest, SetPerCallCredentials) {
+  ResetStub();
+  EchoRequest request;
+  EchoResponse response;
+  ClientContext context;
+  std::shared_ptr<Credentials> creds =
+      IAMCredentials("fake_token", "fake_selector");
+  context.set_credentials(creds);
+  grpc::string msg("Hello");
+
+  Status s = stub_->Echo(&context, request, &response);
+  // TODO(yangg) verify creds at the server side.
+  EXPECT_EQ(request.message(), response.message());
+  EXPECT_TRUE(s.IsOk());
+}
+
+TEST_F(End2endTest, InsecurePerCallCredentials) {
+  ResetStub();
+  EchoRequest request;
+  EchoResponse response;
+  ClientContext context;
+  std::shared_ptr<Credentials> creds = InsecureCredentials();
+  context.set_credentials(creds);
```

Comment:
Does it really make sense here? Shouldn't we fail (maybe later) when an insecure credentials is set on a call?